pkg/alertmanager/amcfg.go

```diff
@@ -135,7 +135,9 @@ func (cb *configBuilder) initializeFromAlertmanagerConfig(ctx context.Context, a
 	}
 
 	// Add routes to globalAlertmanagerConfig.Route without enforce namespace
-	globalAlertmanagerConfig.Route = cb.convertRoute(amConfig.Spec.Route, crKey)
+	if amConfig.Spec.Route != nil {
```

Comment:
Instead of checking route here if check for route nil as first statement in `convertRoute` method would be better

```go
if in == nil  {
    return nil
}
```

https://github.com/prometheus-operator/prometheus-operator/blob/5519586b0ab25e83616e42163e095797973be027/pkg/alertmanager/amcfg.go#L330 

and update check here as

```go
if convertedRoute := cb.convertRoute(amConfig.Spec.Route, crKey); convertedRoute != nil 
    globalAlertmanagerConfig.Route = convertedRoute
}
```

Reply:
Done